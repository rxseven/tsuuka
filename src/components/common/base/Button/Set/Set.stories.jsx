import React from 'react';
import { storiesOf } from '@storybook/react';

import Set from './index';

const title = 'Components/Common/Base/Button/Set';

storiesOf(title, module).add('default', () => (
  <Set>
    <button type="button">Button</button>
    <button type="button">Button</button>
  </Set>
));
