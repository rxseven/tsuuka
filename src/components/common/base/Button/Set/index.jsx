import PropTypes from 'prop-types';
import exact from 'prop-types-exact';
import React from 'react';
import styled from 'styled-components';

const Frame = styled.div`
  align-items: center;
  display: flex;

  > :not(:last-child) {
    margin-right: 0.25rem;
  }

  > :not(:first-child) {
    margin-left: 0.25rem;
  }
`;

const propTypes = exact({
  children: PropTypes.node.isRequired
});

function Set({ children }) {
  return <Frame>{children}</Frame>;
}

Set.propTypes = propTypes;

export default Set;
