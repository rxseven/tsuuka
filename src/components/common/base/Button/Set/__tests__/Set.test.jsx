import { shallow } from 'enzyme';
import React from 'react';
import renderer from 'react-test-renderer';

import Set from '../index';

describe('<Button.Set />', () => {
  // Arrange
  const component = (
    <Set>
      <button type="button">Button</button>
      <button type="button">Button</button>
    </Set>
  );

  describe('Unit tests', () => {
    it('should render without crashing', () => {
      // Act
      const wrapper = shallow(component);

      // Assert
      expect(wrapper).toBeDefined();
    });
  });

  describe('Snapshot tests', () => {
    it('should render correctly', () => {
      // Act
      const tree = renderer.create(component).toJSON();

      // Assert
      expect(tree).toMatchSnapshot();
    });
  });
});
