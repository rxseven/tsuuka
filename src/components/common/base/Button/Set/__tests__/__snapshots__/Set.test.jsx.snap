// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button.Set /> Snapshot tests should render correctly 1`] = `
<div
  className="sc-bdVaJa fYgCuF"
>
  <button
    type="button"
  >
    Button
  </button>
  <button
    type="button"
  >
    Button
  </button>
</div>
`;
